pyscf/nao/gw.py

```diff
@@ -311,30 +313,32 @@ def report(self):
       print("\f   n  %14s %14s %7s " % ("E_mf", "E_gw", "occ") )
       for ie,(emf,egw,f) in enumerate(zip(emfev,egwev,self.mo_occ[0].T)):
         print("%5d  %14.7f %14.7f %7.2f " % (ie, emf[0], egw[0], f[0]) )
-      print('\fG0W0 HOMO energy    (eV): {:.7f}'.format(egwev[self.nfermi[0]-1,0]))
-      print('G0W0 LUMO energy    (eV): {:.7f}'.format(egwev[self.nfermi[0],0]))
-      print('G0W0 HOMO-LUMO gap  (eV): {:.7f}'.format(egwev[self.nfermi[0],0]-egwev[self.nfermi[0]-1,0]))
+      print('Fermi energy        (eV):\t{:f}'.format(self.fermi_energy* HARTREE2EV))
+      print('G0W0 HOMO energy    (eV):\t{:f}'.format(egwev[self.nfermi[0]-1,0]))
+      print('G0W0 LUMO energy    (eV):\t{:f}'.format(egwev[self.nfermi[0],0]))
+      print('G0W0 HOMO-LUMO gap  (eV):\t{:f}'.format(egwev[self.nfermi[0],0]-egwev[self.nfermi[0]-1,0]))
     elif self.nspin==2:
       print("\f    n %14s %14s  %7s | %14s %14s  %7s" % ("E_mf_up", "E_gw_up", "occ_up", "E_mf_down", "E_gw_down", "occ_down") )
       for ie,(emf,egw,f) in enumerate(zip(emfev,egwev,self.mo_occ[0].T)):
         print("%5d  %14.7f %14.7f %7.2f | %14.7f %14.7f %7.2f" % (ie, emf[0], egw[0], f[0],  emf[1], egw[1], f[1]) )
       print('')
-      print('\fG0W0 HOMO energy    (eV): {:.7f}\t{:.7f}'.format(egwev[self.nfermi[0]-1,0],egwev[self.nfermi[1]-1,1]))
-      print('G0W0 LUMO energy    (eV): {:.7f}\t{:.7f}'.format(egwev[self.nfermi[0],0],egwev[self.nfermi[1],1]))
-      print('G0W0 HOMO-LUMO gap  (eV): {:.7f}\t{:.7f}'.format(egwev[self.nfermi[0],0]-egwev[self.nfermi[0]-1,0],egwev[self.nfermi[1],1]-egwev[self.nfermi[1]-1,1]))
+      print('Fermi energy        (eV):\t{:f}'.format(self.fermi_energy* HARTREE2EV))
+      print('G0W0 HOMO energy    (eV):\t{:f}\t{:f}'.format(egwev[self.nfermi[0]-1,0],egwev[self.nfermi[1]-1,1]))
+      print('G0W0 LUMO energy    (eV):\t{:f}\t{:f}'.format(egwev[self.nfermi[0],0],egwev[self.nfermi[1],1]))
+      print('G0W0 HOMO-LUMO gap  (eV):\t{:f}\t{:f}'.format(egwev[self.nfermi[0],0]-egwev[self.nfermi[0]-1,0],egwev[self.nfermi[1],1]-egwev[self.nfermi[1]-1,1]))
     else:
       raise RuntimeError('not implemented...')
     t= int(timer() - starting_time)
-    print('\fTotal running time is: {:02d}:{:02d}:{:02d}'.format(t // 3600, (t % 3600 // 60), t % 60))
+    print('\fTotal running time is: {:02d}:{:02d}:{:02d}\nJOB DONE!'.format(t // 3600, (t % 3600 // 60), t % 60))
     
   def make_mo_g0w0(self):
     """ This creates the fields mo_energy_g0w0, and mo_coeff_g0w0 """
 
     self.h0_vh_x_expval = self.get_h0_vh_x_expval()
     if self.verbosity>1:
-      np.set_printoptions(linewidth=1000, suppress=True, precision=5)
-      print(__name__, '.h0_vh_x_expval: ')
-      print(self.h0_vh_x_expval* HARTREE2EV)
+      print(__name__,'\t\t====> h0_vh_x_expval:\n no.\t  H_up\t\t  H_down')
```

Comment:
Great!

Reply:
Could you share a better definition of 'h0_vh_x_expval'?